public/app/features/alerting/unified/GrafanaRuleQueryViewer.tsx

```diff
@@ -98,56 +99,110 @@ export function GrafanaRuleQueryViewer({
 }
 
 interface QueryPreviewProps extends Pick<AlertQuery, 'refId' | 'relativeTimeRange' | 'model'> {
+  rule: CombinedRule;
   isAlertCondition: boolean;
   dataSource?: DataSourceInstanceSettings;
   queryData?: PanelData;
   thresholds?: ThresholdDefinition;
-  evalTimeRange?: RelativeTimeRange;
-  onEvalTimeRangeChange: (timeRange: RelativeTimeRange) => void;
 }
 
 export function QueryPreview({
   refId,
-  relativeTimeRange,
+  rule,
   thresholds,
   model,
   dataSource,
   queryData,
-  evalTimeRange,
-  onEvalTimeRangeChange,
+  relativeTimeRange,
 }: QueryPreviewProps) {
   const styles = useStyles2(getQueryPreviewStyles);
+  const isExpression = isExpressionQuery(model);
+  const [exploreSupported, exploreAllowed] = useAlertRuleAbility(rule, AlertRuleAction.Explore);
+  const canExplore = exploreSupported && exploreAllowed;
+
+  const headerItems: React.ReactNode[] = [];
+
+  if (dataSource) {
+    const dataSourceName = dataSource.name ?? '[[Data source not found]]';
+    const dataSourceImgUrl = dataSource.meta.info.logos.small;
+
+    headerItems.push(<DataSourceBadge name={dataSourceName} imgUrl={dataSourceImgUrl} key="datasource" />);
+  }
 
-  // relativeTimeRange is what is defined for a query
-  // evalTimeRange is temporary value which the user can change
-  const headerItems = [dataSource?.name ?? '[[Data source not found]]'];
   if (relativeTimeRange) {
-    headerItems.push(mapRelativeTimeRangeToOption(relativeTimeRange).display);
+    headerItems.push(
+      <Text color="secondary" key="timerange">
+        {secondsToHms(relativeTimeRange.from)} to now
+      </Text>
+    );
+  }
+
+  let queryPreview: JSX.Element = (
+    <pre>
+      <code>{dump(model)}</code>
+    </pre>
+  );
+
+  if (dataSource?.type === DataSourceType.Prometheus && isPromOrLokiQuery(model)) {
+    queryPreview = <PrometheusQueryPreview query={model.expr} />;
+  }
+
+  if (dataSource?.type === DataSourceType.Loki && isPromOrLokiQuery(model)) {
+    queryPreview = <LokiQueryPreview query={model.expr ?? ''} />;
+  }
+
```

Comment:
How about extracting this into a separate component? 
It would be nice if adding a new read-only viewer was a very simple change
It would be awesome to have metrics about the most common types of data sources in alert rules